controller/src/main/java/com/epam/esm/controller/CertificateController.java

```diff
@@ -0,0 +1,130 @@
+package com.epam.esm.controller;
+
+import com.epam.esm.dto.CertificateDTO;
+import com.epam.esm.dto.PatchDTO;
+import com.epam.esm.dto.QuerySpecificationDTO;
+import com.epam.esm.exception.IdInvalidException;
+import com.epam.esm.exception.InvalidDataFormException;
+import com.epam.esm.service.CertificateService;
+import lombok.AllArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+import java.util.List;
+
+/**
+ * The type Certificate controller.
+ */
+@RestController
+@RequestMapping(value = "/certificates")
+@AllArgsConstructor
+public class CertificateController {
+
+    private final CertificateService service;
+
+    /**
+     * finds all certificates by criteria
+     *
+     * @param querySpecificationDTO the query specification dto
+     * @return List of all certificates
+     */
+    @GetMapping
+    public List<CertificateDTO> findAllCertificatesByCriteria(QuerySpecificationDTO querySpecificationDTO) {
```

Comment:
I would use `findAllByCriteria()` for `CertificateController` class and same for `create()`, `update`, `patchUpdate()` and `delete()` methods. Please review also `TagController` class.